src/components/navigation/LinkBase.js

```diff
@@ -0,0 +1,30 @@
+import { downcastRef } from '../../util/typing';
+
+/**
+ * @typedef {import('../../types').BaseProps} BaseProps
+ * @typedef {import('preact').JSX.HTMLAttributes<HTMLAnchorElement>} HTMLAnchorAttributes
+ */
+
+/**
+ * Base component for Link components. Applies common attributes.
+ *
+ * @param {BaseProps & HTMLAnchorAttributes} props
+ */
+export default function LinkBase({
```

Comment:
So, `LinkBase` is responsible for attributes (or any other common functionality) and the components that use it are handling styling.